extensions/cornerstone/src/components/ViewportWindowLevel/getViewportVolumeHistogram.ts

```diff
@@ -32,12 +28,13 @@ const getViewportVolumeHistogram = async (viewport, volume, options?) => {
 
   let scalarData = volume.scalarData;
 
-  let prevTimePoint;
   if (volume.numTimePoints > 1) {
-    prevTimePoint = volume.timePointIndex;
-    const middleTimePoint = Math.round(volume.numTimePoints / 2);
-    volume.timePointIndex = middleTimePoint;
-    scalarData = volume.getScalarData(middleTimePoint);
+    const targetTimePoint = volume.numTimePoints - 1; // or any other time point you need
+    scalarData = volume.voxelManager.getTimePointScalarData(targetTimePoint);
+  }
+
+  if (!scalarData?.length) {
+    return undefined;
   }
 
   const { dimensions, origin, direction, spacing } = volume;
```

Comment:
Just below the range should exclude pixel padding values, for which you should pass in the pixel padding range data.

Reply:
Seems like we need a proper fix for padding values here and there